diceRoll/diceRollUnitTests.js

```diff
@@ -65,63 +38,118 @@ function assert(condition, message) {
   }
 }
 
+function stub() {
+  var info = {
+    called: {
+      true: false,
+      expectedCount: null,
+      count: 0,
+      check: function(expectedCount) {
+        info.called.expectedCount = expectedCount;
+        if (info.called.expectedCount === undefined || info.called.expectedCount === null) {
+          return info.called.true;
+        } else {
+          return (info.called.count === info.called.expectedCount);
+        }
+      }
+    },
+    args: {
+      expected: null,
+      current: null,
+      all: {
+        array: [],
+        check: function(index, expectedArray) {
+          var index = arguments[0];
+          info.args.expected = {};
+          for (key in arguments) {
+            if (key !== "0") {
+              info.args.expected[key - 1] = arguments[key];
+            }
+          }
+          return (JSON.stringify(info.args.expected) === JSON.stringify(info.args.all.array[index]));
+        }
+      },
+      log: function(current) {
+        info.args.all.array[info.called.count] = current;
+      },
+      check: function(expectedArgs) {
+        info.args.expected = arguments;
+        return (JSON.stringify(info.args.expected) === JSON.stringify(info.args.current));
+      }
+    },
+  }
+
+  var placeHolder = function() {
+    info.called.true = true;
```

Comment:
I see why, but I wouldnt have a property named true that is sometimes false... 

You can just use a single "called" count property and if it is greater than 0 then you know it has been called.

info.called++;

And later

return info.called > 0;

Reply:
Sorry, I can see you are using called as an on. I meant use:

info.called.count++;
And check method can treat this as true:
return info.called.count > 0;